metadata/client.go

```diff
@@ -1470,8 +1472,20 @@ func (variant *FeatureVariant) Properties() Properties {
 	return variant.fetchPropertiesFn.Properties()
 }
 
-func (variant *FeatureVariant) Category() FeatureVariantCategory {
-	return FeatureVariantCategory(variant.serialized.GetCategory())
+func (variant *FeatureVariant) Mode() ComputationMode {
+	return ComputationMode(variant.serialized.GetMode())
+}
+
+func (variant *FeatureVariant) IsOnDemand() bool {
```

Comment:
can we use `variant.serialized.GetIsOnDemand()` or something similar here? I am just afraid that if we add more modes, then we will need to make sure to update this too.